releases/__init__.py

```diff
@@ -598,8 +598,8 @@ def unknown_visit(self, node):
 
 def generate_changelog(app, doctree):
     # Don't scan/mutate documents that don't match the configured document name
-    # (which by default is changelog.rst).
-    if app.env.docname != app.config.releases_document_name:
+    # (which by default is ['changelog.rst', ]).
+    if app.env.docname not in app.config.releases_document_name:
```

Comment:
Offhand I do prefer to leverage the fact that `conf.py` is a `.py`, so +1 to "just ask user to supply a list". Or the user could use `glob.glob` from stdlib, or third party modules like `pathlib` as noted.

tl;dr when a use case is uncommon-to-rare, enabling it via "user is enabled to write their own tiny bit of Python" is preferable over "we write our own 'blessed' tiny-bit-of-Python and expose as a func/feature".

So as long as we allow a list value in the setting, the user is free to generate it however they want, and we're neither assisting nor limiting them.